### PR TITLE
Add documentation for lax.stop_gradient.

### LINKED ISSUE #492: Document `lax.stop_gradient`
Is there a function similar to tensorflow's [tf.stop_gradient](https://www.tensorflow.org/api_docs/python/tf/stop_gradient)?

How can I do this in jax??

### PR DESCRIPTION
**Pull request by hawkinsp**
Fixes #492

### CODE REVIEW

File: jax/lax.py [hunk 1]
```diff
@@ -1114,6 +1114,19 @@ def broadcasted_eye(dtype, shape, axes):
 
 
 def stop_gradient(x):
+  """Stops gradient computation.
+
+   Operationally `stop_gradient` is the identity function, that is, it returns
+   argument `x` unchanged. However, `stop_gradient` prevents the flow of
+   gradients during forward or reverse-mode automatic differentiation.
```
Comment (mattjj) [inline jax/lax.py]: We could add an explanation of how it works with higher-order differentiation: does it stop just "one level" of differentiation, or all levels? (It's the latter: it stops all levels, though it's easy to add a function that does the latter if we ever want it.)

Brainstorming another angle of explanation, here are two ways to define the same function `gradfun`:

```python
# first way, using lax.stop_gradient
def f1(x):
  return lax.sin(x) * lax.cos(lax.stop_gradient(x))
gradfun = grad(f1)

# second way, without lax.stop_gradient
def f2(x, y):
  return lax.sin(x) * lax.cos(y)
gradfun = lambda x: grad(f2)(x, x)
```

In words, `lax.stop_gradient` is really just a shortcut way to express a function with two inputs, differentiating with respect to one of them and evaluating both arguments at the same point.

Hmm maybe that's more confusing... Treat that as brainstorming, but we should probably specify the higher-order behavior.

**Review by mattjj — APPROVED** [review]
Nice!

File: jax/lax.py (hunk 1 above)
Comment (hawkinsp) [reply jax/lax.py]: Decided to leave the text as is, but clarified gradients are stopped at all levels, and gave an example thereof.
